packages/connect/src/Detector.ts

```diff
@@ -159,31 +134,34 @@ export class Detector {
    * @remarks
    * This is used internally for advanced PolkadotJS use cases and is not supported.  Use {@link connect} instead.
    */
-  public provider = (
+  public provider = async (
     chain: ChainInfo,
     parachainSpec?: string,
-  ): ProviderInterface => {
-    let provider: ProviderInterface = {} as ProviderInterface
-
-    if (!chain.name && !Object.keys(this.#chainSpecs).includes(chain.name)) {
+  ): Promise<ProviderInterface> => {
+    if (!chain.name && !SupportedChains[chain.name as SupportedChains]) {
       throw new Error(
         `No known Chain was detected and no chainSpec was provided. Either give a known chain name ('${Object.keys(
-          this.#chainSpecs,
+          SupportedChains,
         ).join("', '")}') or provide valid chainSpecs.`,
       )
     }
 
     if (this.#isExtension) {
-      provider = new ExtensionProvider(
+      const { ExtensionProvider } = await import(
+        "./ExtensionProvider/ExtensionProvider.js"
+      )
```

Comment:
I don't think we need this one, do we? The extension provider is supposed to be very small.